creusot-contracts/src/ghost/atomic_invariant.rs

```diff
@@ -0,0 +1,90 @@
+//! Define atomic invariants.
+//!
+//! Atomic invariants are used to specify concurrent operations.
+
+use crate::{ghost::FnGhost, prelude::*};
+
+/// Creusot wrapper around [`std::sync::atomic::AtomicI32`]
+pub struct AtomicI32(::std::sync::atomic::AtomicI32);
+
+/// Does a single atomic operation, in ghost
+#[opaque]
+pub struct Committer {}
+
+/// Ownership for a [`AtomicI32`].
+#[opaque]
+pub struct AtomicI32Own {}
+
+impl AtomicI32Own {
+    /// Which atomic does this holds the ownership for
+    #[logic(opaque)]
+    pub fn id(self) -> AtomicI32 {
+        dead
+    }
+
+    /// Value currently contained in the atomic
+    #[logic(opaque)]
+    pub fn value(self) -> i32 {
```

Comment:
```suggestion
    pub fn val(self) -> i32 {
```

Coherence with e.g., `PermCell`.